text/0000-debug-noop-in-release-builds.md

```diff
@@ -0,0 +1,30 @@
+- Feature Name: debug-noop-in-release
+- Start Date: 2014-03-04
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+The `{:?}` format specifier should not print values in release builds.
+
+# Motivation
+
+Many commonly used methods, such as `unwrap`, use `{:?}` to print out representations of values if their conditions are not met. This is useful for debugging but is not so useful in release builds, and it can bloat up the generated binaries. Worse, it adds opaque calls (as `Debug` is frequently a lot of code and rarely gets inlined), which can mess up the inlining heuristics and cause cascades of missed optimization opportunities.
```

Comment:
Well, the problem here is `panic!` in particular. Maybe we shouldn't be saving panic messages in release builds...